app/models/puppetdb_client.rb

```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+require 'puppetdb'
+
+class PuppetdbClient
+  def initialize(endpoint)
+    @endpoint = endpoint
+    @client = PuppetDB::Client.new(server: @endpoint)
```

Comment:
See my comment on the `initialize` method for the `ServerPuppetData` class but for GithubClient class we initialize the client here rather than in the class that initializes this class.